pages/your_tasks/virtual_dev_environments.md

```diff
@@ -0,0 +1,120 @@
+able to keep these configurations separate to avoid confusion between projects. 
+The solution for this problem is to create a self-contained virtual environment per project, which contains a particular version of your 
+programming language interpreter/compiler plus a number of additional external libraries.
+
+Another big motivator for using virtual environments is that they make **sharing your code with others** (users or developers) much easier.
+By sharing a description of your virtual development environment you enable others to quickly replicate the same environment 
+on their machines and run or further develop your software - making your work **portable**, **reusable** and more **reproducible**.
+
+### Considerations
+
+- As more external libraries are added to your software project over time, you can add them to its specific virtual environment
+and avoid a great deal of confusion by having separate (smaller) virtual environments for each project
+rather than one huge global environment on your machine with potential package version clashes.
+- You have an older project that only works under, e.g., Python 2. You do not have the time to migrate the project to Python 3
+or it may not even be possible as some of the third party dependencies are not available under Python 3.
+You have to start another project under Python 3. The best way to do this on a single machine is
+to set up two separate Python virtual environments.
+- One of your Python 3 projects is locked to use a particular older version of a third party dependency.
+You cannot use the latest version of the dependency as it breaks things in your project.
+In a separate branch of your project, you want to try and fix problems introduced by the new version of the dependency
+without affecting the working version of your project. You need to set up a separate virtual environment for your branch to
+'isolate' your code while testing the new feature.
+- You do not have to worry too much about specific versions of external libraries that your project depends on most of the time.
+Virtual environments enable you to always use the latest available version without specifying it explicitly.
+They also enable you to use a specific older version of a package for your project, should you need to. 
+
+### Solutions
+
+- Make your research software reusable and your research that relies on that software reproducible by setting up and sharing its virtual development environment.
+
+## How do you create virtual software development environments? 
+
+### Description
+
+Most modern programming languages use some kind of virtual environments or a similar mechanism to isolate libraries or dependencies for a specific project, 
+making it easier to develop, run, test and share code with others. 
+Some examples include Bundler for Ruby, Conan for C++, or Maven with classpath for Java. 
```

Comment:
We mentioned python a lot - so might be worth mentioning the one for python here (eps. given it's popularity)